### PR TITLE
Added definition for tough-cookie-filestore

### PR DESCRIPTION
**Pull request by friedow**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Adding a new definition:
- [x] The package does not already provide its own types
- [x] If this is for an NPM package, match the name.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/tough-cookie-filestore/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// Type definitions for tough-cookie-filestore 0.0
+// Project: https://github.com/mitsuru/tough-cookie-filestore
+// Definitions by: Christian Friedow <https://github.com/friedow>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare namespace FileCookieStore {
+    export interface Cookie {
```
Comment (uniqueiniquity) [inline types/tough-cookie-filestore/index.d.ts]: Don't need the keyword `export` here.

File: types/tough-cookie-filestore/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    removeCookie(domain: string, path: string, key: string, cb: (whatever: null) => void): void;
+
+    removeCookies(domain: string, path: string, cb: (whatever: null) => void): void;
+
+    saveToFile(filePath: string, data: any, cb: () => void): void;
+
+    loadFromFile(filePath: string, cb: (data: any) => void): void;
+}
+
+export = FileCookieStore
```
Comment (uniqueiniquity) [inline types/tough-cookie-filestore/index.d.ts]: Need a semicolon at the end of this line

File: types/tough-cookie-filestore/tough-cookie-filestore-tests.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import FileCookieStore from 'tough-cookie-filestore';
```
Comment (uniqueiniquity) [inline types/tough-cookie-filestore/tough-cookie-filestore-tests.ts]: This should work if you enable `esModuleInterop` in the tsconfig.json file. Alternatively, use `import FileCookieStore = 'tough-cookie-filestore';` in this test, and just make sure you use `esModuleInterop` in your consuming code.

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]

File: types/tough-cookie-filestore/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
+// Type definitions for tough-cookie-filestore 0.0
+// Project: https://github.com/mitsuru/tough-cookie-filestore
+// Definitions by: Christian Friedow <https://github.com/friedow>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export = FileCookieStore;
+export as namespace FileCookieStore;
```
Comment (uniqueiniquity) [inline types/tough-cookie-filestore/index.d.ts]: This line should be removed; as far as I can tell, `tough-cookie-filestore` is not a UMD module.

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]
Just one small thing to remove, but looks good otherwise!

File: types/tough-cookie-filestore/index.d.ts (hunk 4 above)
Comment (uniqueiniquity) [reply types/tough-cookie-filestore/index.d.ts]: That is, `export as namespace` is only relevant when the underlying module is a UMD module.

Comment (friedow) [reply types/tough-cookie-filestore/index.d.ts]: Changed that, thanks for the explanation 👍 .

**Review by uniqueiniquity — APPROVED** [review]

**Comment by uniqueiniquity** [comment]
Congratulations on your first DefinitelyTyped contribution!
Thank you for being a part of the community!

**Comment by friedow** [comment]
Very nice :)
Thanks a lot for the detailed review!

**Pull request merged**